File: app/src/main/java/com/example/foodrecipes/viewmodels/CompanyInfoViewModel.java
package com.example.foodrecipes.viewmodels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.foodrecipes.model.CompanyInfo;
import com.example.foodrecipes.repositories.CompanyRepository;

public class CompanyInfoViewModel extends ViewModel {

    private CompanyRepository mCompanyRepository;

    public CompanyInfoViewModel(){
        mCompanyRepository = CompanyRepository.getInstance();
    }

    public LiveData<CompanyInfo> getCompany(){
        return mCompanyRepository.getCompany();
    }

    public void getCompanyInfo(String symbol){
        mCompanyRepository.getCompanyInfo(symbol);
    }

    public LiveData<Boolean> isUpdating(){
        return mCompanyRepository.isUpdating();
    }

    public LiveData<Boolean> hasRetrieveError(){
        return mCompanyRepository.hasRetrieveError();
    }

    public void setHasNotRetrieveError(){
        mCompanyRepository.setHasNotRetrieveError();
    }
}

File: app/src/main/java/com/example/foodrecipes/viewmodels/CompanyInfoViewModel.kt
package com.example.foodrecipes.viewmodels

import androidx.hilt.Assisted
import androidx.hilt.lifecycle.ViewModelInject
import androidx.lifecycle.*
import com.example.foodrecipes.model.CompanyInfo
import com.example.foodrecipes.repositories.CompanyRepositoryImpl
import com.example.foodrecipes.util.Resource
import kotlinx.coroutines.ExperimentalCoroutinesApi
import kotlinx.coroutines.flow.launchIn
import kotlinx.coroutines.flow.onEach
import kotlinx.coroutines.launch

class CompanyInfoViewModel
@ViewModelInject
constructor(
        private val repository: CompanyRepositoryImpl,
        @Assisted private val savedStateHandle: SavedStateHandle
): ViewModel(){

    private val _company: MutableLiveData<Resource<CompanyInfo>> = MutableLiveData()
    val company: LiveData<Resource<CompanyInfo>>
        get() = _company

    @ExperimentalCoroutinesApi
    fun getCompanyInfo(symbol: String){
        viewModelScope.launch {
            repository.getCompanyInfo(symbol).onEach { resource ->
                _company.value = resource
            }.launchIn(viewModelScope)
        }
    }
}